packages/editor/src/lib/exports/ExportDelay.tsx

```diff
@@ -0,0 +1,47 @@
+import { bind } from '@tldraw/utils'
+
+export class ExportDelay {
+	private isResolved = false
+	private readonly promisesToWaitFor: Promise<void>[] = []
+
+	constructor(private readonly maxDelayTimeMs: number) {}
+
+	@bind waitUntil(promise: Promise<void>): void {
+		// console.log('waitUntil', new Error().stack)
+		if (this.isResolved) {
+			throw new Error(
+				'Cannot `waitUntil` - the export has already been resolved. Make sure to call `waitUntil` as soon as possible during an export - ie within the first react effect after rendering.'
```

Comment:
this comment makes me think an example would be good. i would have expected the AG grid example to have a waitUntil but it looks like it's not in there? i thought you had mentioned the AG grid required it but maybe I misheard

Reply:
`waitUntil` is a bit low-level. The docs for it refer you to `useDelaySvgExport`, which is built on top of it and is what AG grid uses. I could mark `waitUntil` as internal for now?

Reply:
ah no i see it now, somehow i had missed it - nevermind, it makes sense as is